testing/classes/TestSuite.lua

```diff
@@ -0,0 +1,183 @@
+TestSuite = {
+
+
+  -- @method - TestSuite:new()
+  -- @desc - creates a new TestSuite object that handles all the tests
+  -- @return {table} - returns the new TestSuite object
+  new = function(self)
+    local test = {
+
+      -- testsuite internals
+      modules = {},
+      module = nil,
+      test = nil,
+      testcanvas = nil,
+      current = 1,
+      output = '',
+      totals = {0, 0, 0},
+      time = 0,
+      xml = '',
+      html = '',
+      mdrows = '',
+      mdfailures = '',
+      delayed = nil,
+      fakequit = false,
+      windowmode = true,
+
+      -- love modules to test
+      audio = {},
+      data = {},
+      event = {},
+      filesystem = {},
+      font = {},
+      graphics = {},
+      image = {},
+      joystick = {},
+      math = {},
+      mouse = {},
+      physics = {},
+      sound = {},
+      system = {},
+      thread = {},
+      timer = {},
+      touch = {},
+      video = {},
+      window = {}
+
+    }
+    setmetatable(test, self)
+    self.__index = self
+    return test
+  end,
+
+
+  -- @method - TestSuite:runSuite()
+  -- @desc - called in love.update, runs through every method or every module
+  -- @param {number} delta - delta from love.update to track time elapsed
+  -- @return {nil}
+  runSuite = function(self, delta)
+
+      -- stagger between tests
+    if self.module ~= nil then
+
+        if self.module.start == true then
```

Comment:
`if foo == true then` can be simplified to `if foo then` (and `if foo == false then` can be `if not foo then`)